src/mbgl/gfx/index_buffer.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <memory>
+
+namespace mbgl {
+namespace gfx {
+
+class IndexBufferResource {
+protected:
+    IndexBufferResource() = default;
+public:
+    virtual ~IndexBufferResource() = default;
+};
+
+class IndexBuffer {
+public:
+    std::size_t elements;
+    std::unique_ptr<const IndexBufferResource> resource;
```

Comment:
Could we use `util::peer` instead (I guess it'd positively influence the binary size change)?